test/test-network/systemd-networkd-tests.py

```diff
@@ -6323,6 +6375,59 @@ def check(self, ipv4, ipv6):
         check(self, False, True)
         check(self, False, False)
 
+    def test_dhcp_client_use_dnr(self):
```

Comment:
As far as I understand the test covers the DHCP options only. I wonder if there is anything that can cover the router advertisements? I took a look at radvd and it seems it's unlikely to support the encrypted DNS option anytime soon. One option could be to send raw packets but that's cumbersome. If networkd could send that option it could probably test itself by sending and receiving that option.

Reply:
>  It's quite difficult to test the ra stuff

Agreed. This thread can probably be resolved.